config.go

```diff
@@ -105,11 +47,11 @@ func ParseProxyURL(val string) (*url.URL, error) {
 
 	creds, hostport, ok := strings.Cut(hostport, "@")
 	if ok {
-		ui, err := ParseUserInfo(creds)
+		ui, err := ParseUserinfo(creds)
 		if err != nil {
 			return nil, err
 		}
-		u.User = ui
+		u.User = url.UserPassword(ui.Username, ui.Password)
```

Comment:
I think this could be moved to Userinfo type like `AsURLUserinfo`, and the type needs some comment why it exists.